packages/govuk-frontend/src/govuk/helpers/typography.unit.test.js

```diff
@@ -241,13 +241,36 @@ describe('@mixin govuk-typography-responsive', () => {
     )
   })
 
+  // govuk-typography-responsive is the previous name for govuk-font-size
+  it('throws a deprecation warning if govuk-typography-responsive is used', async () => {
```

Comment:
Can we also test that `govuk-typography-responsive` still does what we expect? i.e. that when all arguments are passed it generates the expected CSS output?

(Could potentially call `govuk-typography-responsive` and `govuk-font` with the same arguments and expect they match?)

Reply:
I've reorganised the `govuk-typography-responsive` testing to cover this. I only tested against the basic `govuk-font-size` test as I was finding testing against `govuk-font` tricky to do without a lot of messy string manipulation. This might be my own misunderstanding of our embedded sass strings.

I don't know if this needs further testing against `govuk-font-size` like if `$line-height` or `$important` are set.

Reply:
Like how you've organised that, makes sense 👍🏻

I do think it'd be most valuable to test with all three arguments so we know they're all being passed as expected. For example, if we just forgot to pass the `$important` param to `govuk-font-size` all our tests would still pass at the mo.

From a quick play I think something like this would work:

```
it('outputs the same CSS as govuk-font-size', async () => {
  const sass = `
    ${sassBootstrap}

    .foo {
      @include govuk-typography-responsive(
        $size: 14,
        $override-line-height: 40px,
        $important: true
      )
    }
  `

  const expectedSass = `
    ${sassBootstrap}

    .foo {
      @include govuk-font-size(
        $size: 14,
        $line-height: 40px,
        $important: true
      )
    }
  `

  await expect(compileSassString(sass)).resolves.toMatchObject({
    css: (await compileSassString(expectedSass)).css
  })
})
```

But I'm not sure if that's the most idiomatic way of comparing the results of two promises – perhaps @colinrotherham can advise?

Reply:
Aha that's a much better idea. I'll have a go at implementing that for the mo.

Reply:
Multiple correct answers!

If concurrent compiling isn't a must there's always:

```mjs
const { css: result1 } = await compileSassString('Example 1')
const { css: result2 } = await compileSassString('Example 2')

expect(result1).toEqual(result2)
```

Or start the promises running early but await them later?

Still got to faff around with that nested CSS property

```mjs
const result1 = compileSassString('Example 1')
const result2 = compileSassString('Example 2')

await expect(result1).resolves.toMatchObject({
  css: (await result2).css
})
```

Or chain them in advance so each promise awaits to CSS in readiness?

But Jest has no [`expectAsync()` like Jasmine](https://jasmine.github.io/api/5.1/global.html#expectAsync) so odd to `.resolves` the 1st promise and `await` the 2nd

```mjs
const css1 = compileSassString('Example 1')
  .then(({ css }) => css)

const css2 = compileSassString('Example 2')
  .then(({ css }) => css)

await expect(css1).resolves.toEqual(await css2)
```

Or combine them into a single `await Promise.all()` and extracting the CSS after?

```mjs
const [css1, css2] = await Promise.all([
  compileSassString('Example 1'),
  compileSassString('Example 2')
]).then((results) => results.map(({ css }) => css))

expect(css1).toEqual(css2)
```

Can use as much or little shorthand as you like:

```patch
- ]).then((results) => results.map(({ css }) => css))
+ ]).then((results) => results.map((result) => result.css))
```

Or split out the `.map()` onto another line if it's clearer?

```mjs
const results = await Promise.all([
  compileSassString('Example 1'),
  compileSassString('Example 2')
])

const [result1, result2] = results.map(({ css }) => css)

expect(result1).toEqual(result2)
```

Reply:
Can't leave out [`for await...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of) but shame we only have one result for comparison:

```mjs
const result = await compileSassString('Example 1')

const results = [
  compileSassString('Example 2'),
  compileSassString('Example 3'),
  compileSassString('Example 4')
]

for await (const { css } of results) {
  expect(css).toEqual(result.css)
}
```

Reply:
Thanks for this Colin! As you said, lots of correct answers. I'm going to take the lazy way out and say lets figure this out closer to the point at which this is going out. I get the impression as a pre-review we're fairly happy with this in prospect.